src/rgw/rgw_lua.cc

```diff
@@ -96,6 +96,11 @@ int add_package(const DoutPrefixProvider *dpp, rgw::sal::Store* store, optional_
   if (p.empty()) {
     return -ECHILD;
   }
+  rgw::sal::RadosStore *rados = dynamic_cast<rgw::sal::RadosStore*>(store);
+  if (!rados) { /* Supported for only RadosStore */
+    return -1;
```

Comment:
> maybe these interfaces should take `rgw::sal::RadosStore*` instead of the generic `Store`? most of these functions are only called from `rgw_admin.cc`, which already assumes the store is a `RadosStore`. only `install_packages()` is called from `rgw_main.cc`, and that call alone could be conditional on a `dynamic_cast`

yes..it needs to be addressed only in rgw_main.  will make it conditional. Thanks!